modules/api_docs/php/api_docs.class.inc

```diff
@@ -0,0 +1,105 @@
+<?php declare(strict_types=1);
+/**
+ * PHP Version 7
+ *
+ * @category Documentation
+ * @package  Main
+ * @author   Xavier Lecours <xavier.lecours@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris-Trunk/
+ */
+namespace LORIS\api_docs;
+
+/**
+ * This class list the schema files of this LORIS instance modules
+ * for the swagger-ui app to present. It is a public module but will only
+ * show specification (schemas) of module that the user can access.
+ *
+ * @category Documentation
+ * @package  Main
+ * @author   Xavier Lecours <xavier.lecours@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris-Trunk/
+ */
+
+class API_Docs extends \NDB_Page
+{
+    /**
+     * Determine whether the user has permission to view this page
+     *
+     * @param \User $user The user whose access is being checked
+     *
+     * @return bool whether the user has access
+     */
+    public function hasAccess(\User $user): bool
+    {
+        return true;
+    }
+
+    /**
+     * Sets the template variable for the schema file to load.
+     *
+     * @return void
+     */
+    function setup()
+    {
+        $baseurl = \NDB_Factory::singleton()
+            ->config()
+            ->getSetting('base');
+
+        /*
+         *  The pattern to find schema.yml files will scan the static directory
+         *  of all modules and project's modules. The pattern is ment to be used by
+         *  the glob() function.
```

Comment:
I don't think this is a reliable way to get the modules. It will include inactive ones and invalid ones.

You should use getActiveModules() on the LorisInstance to get a list of active modules and then do a file_exists check on `$moduledir/static/schema.yml`. To do that you may need to convert this from the `setup` function to the `loadResources` function so that you have access to the request object to get the LorisInstance object from the 'loris' attribute on.

It looks like the `dir` property of the module also isn't exposed from the Module class, but it's passed in the constructor and stored in a property, so you can add a getter.

Reply:
I will work on that. Thanks

Reply:
@driusan , I will probably add `getSchemaUrl` to the modules instead.

Reply:
Not opposed to the concept but can you think of a different name? SchemaURL sounds like it's for the SQL schema.

Reply:
@driusan , I added hasAPISpecification in the module class(es)